lib/components/Atom/WheelPicker/index.tsx

```diff
@@ -0,0 +1,231 @@
+import clsx from "clsx";
+import "./wheel-picker.scss";
+import { InputProps } from "@components/Input/base";
+
+export interface WheelPickerProps extends HTMLAttributes<HTMLElement> {
+    data: { value: string | number }[];
+    selectedValue: string | number;
+    setSelectedValue: (value: string | number) => void;
+    handleKeyDown: (e: React.KeyboardEvent) => void;
+    focus: boolean;
+    dropDownItemClassName?: string;
+    containerClassName?: string;
+    listClassName?: string;
+}
+
+export const WheelPicker = ({
+    data,
+    selectedValue,
+    setSelectedValue,
+    handleKeyDown,
+    focus,
+    dropDownItemClassName,
+    containerClassName,
+    listClassName,
+    ...rest
+}: WheelPickerProps) => {
+    const dataRefs = useRef<HTMLDivElement[]>([]);
+    const dataItemsMap = useMemo(
+        () =>
+            data.reduce(
+                (map, item, index) => map.set(item.value, index),
+                new Map(),
+            ),
+        [data],
+    );
+
+    const currentDataValue = useRef(dataItemsMap.get(selectedValue) ?? 0);
+    const dataItemsContRef = useRef<HTMLUListElement>(null);
+
+    const handleDataScroll = (event: Event) => {
+        let isAnimating = false;
+        if (!isAnimating) {
+            isAnimating = true;
+            if (!event.target) return;
+            const scrollTop = Math.max(
+                (event.target as HTMLUListElement).scrollTop,
+                0,
+            );
+            const selectedElement = Math.min(
+                Math.max(Math.floor(scrollTop / 48), 0),
+                data.length - 1,
+            );
+            (dataRefs.current[selectedElement] as HTMLDivElement)?.focus({
+                preventScroll: true,
+            });
+            setSelectedValue(data[selectedElement].value);
+            isAnimating = false;
+        }
+    };
+
+    const resizeObserver = new ResizeObserver(() => {
+        const index = Number(currentDataValue.current);
+        if (!dataItemsContRef?.current) return;
+        const divHeight = dataItemsContRef?.current.clientHeight;
+        dataItemsContRef.current.style.paddingTop = `${divHeight * 0.5}px`;
+        dataItemsContRef.current.style.paddingBottom = `${divHeight * 0.5}px`;
+        if (dataRefs.current[index]) {
+            (dataRefs.current[index] as HTMLDivElement).scrollIntoView({
+                block: "center",
+            });
+        }
+    });
+
+    useEffect(() => {
+        if (!dataItemsContRef.current) return;
+        setSelectedValue(data[currentDataValue.current].value);
+        const divHeight = dataItemsContRef?.current.clientHeight;
+        resizeObserver.observe(dataItemsContRef.current);
+        dataItemsContRef.current.style.paddingTop = `${divHeight * 0.5}px`;
+        dataItemsContRef.current.style.paddingBottom = `${divHeight * 0.5}px`;
+        dataItemsContRef.current.addEventListener("scroll", handleDataScroll);
+        return () => {
```

Comment:
I think we need to `unobserve` or `disconnect` from `dataItemsContRef.current` here too

Reply:
have removed the listeners and disconnected from the observer inside useEffect return